lib/Driver/Driver.cpp

```diff
@@ -2335,6 +2353,15 @@ Job *Driver::buildJobsForAction(Compilation &C, const JobAction *JA,
     llvm::outs() << '\n';
   }
 
+  /* SR-5860: At this point we still have access to the "-e" arguments
+   (via C.getArgs()).
+   
+   J->getArguments() contains the arguments to be passed on into
+   driver::performFrontend(...) in driver.cpp.
+   
+   What I'm not grasping is how to copy those arguments in Compilation into
+   the Job we're returning here.
```

Comment:
The place to do this is ToolChains.cpp, which is responsible for building Job invocations.